src/explore-education-statistics-admin/src/components/form/FormEditor.tsx

```diff
@@ -306,15 +313,30 @@ const FormEditor = ({
           }}
         >
           {process.env.NODE_ENV !== 'test' ? (
-            <CKEditor
-              editor={Editor}
-              config={config}
-              data={value}
-              onChange={handleChange}
-              onFocus={toggleFocused.on}
-              onBlur={handleBlur}
-              onReady={handleReady}
-            />
+            <>
+              <CKEditor
+                editor={Editor}
+                config={config}
+                data={value}
+                onChange={handleChange}
+                onFocus={toggleFocused.on}
+                onBlur={handleBlur}
+                onReady={handleReady}
+              />
+              <Modal
+                title="Insert glossary link"
+                open={showGlossaryModal}
+                onExit={toggleGlossaryModal.off}
+              >
+                <InsertGlossaryItemForm
+                  onCancel={toggleGlossaryModal.off}
+                  onSubmit={item => {
+                    glossaryPlugin.current?.addGlossaryItem(item);
```

Comment:
Upon adding a glossary link, we can see the link and interact with it in the read-only content block:

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/9917868/1457ea77-0cbd-40e1-9fcb-587650d2f01d)

It might be a good idea to render this like the actual preview where `GlossaryEntryButton` is rendered instead for parity i.e.

![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/9917868/048d09d3-4935-4fbf-8847-1756bb64e13c)

Reply:
Clicking on the read only blocks is meant to open the editor, but currently when you click on a link it'll open that link instead, which probably isn't what you'd want as it will take you away from editing the release. Adding `GlossaryEntryButton` means that it'll flash up the modal briefly then open the editor. 

So I've done this:
- prevented links from opening in read only blocks, so if you click on a link it'll open the editor like when you click anywhere else in the block.
- added `GlossaryEntryButton` so the icon shows but disabled the modal from opening, so when you click on them it'll open the editor. 

What do you think?

Reply:
I think there's some accessibility problems with this approach. 

Screen reader users will still be able to tab to `GlossaryEntryButton` and the screen reader will announce that it's a button that you can show a glossary term definition for. This could be quite confusing when it actually does something completely different by changing the block to editing mode.

A similar thing will happen with links in that the screen reader will announce them, but the behaviour will not be what they're expecting.

There's two options from what I can see:

1. Leave interactive elements (buttons, links, etc) to work normally and don't toggle to editing mode if they directly interacted with one of these
2. Make any interactive elements within the preview container non-interactive

Option 2 might be preferable if we want to streamline the editing experience a bit more. There's a whole bunch of techniques for implementing it, but the simplest seems to be to add the `inert` attribute to a parent element. Seems like support is pretty good across browsers nowadays.

Reply:
Yeah good point. I've gone for option 2 using `inert` (with a workaround because of https://github.com/facebook/react/issues/17157)